engines/advancedDetector.cpp

```diff
@@ -505,57 +540,85 @@ namespace Common {
 }
 
 // Sync with engines/game.cpp
-static char flagsToMD5Prefix(uint32 flags) {
-	if (flags & ADGF_MACRESFORK) {
-		if (flags & ADGF_TAILMD5)
+static char flagsToMD5Prefix(FilePropertiesFlags flags) {
```

Comment:
Why introduce this duplicate thing? Now you suddenly need to translate between these two sets of equivalent flags, one being a superset of another.